include/mbedtls/cipher.h

```diff
@@ -336,11 +336,12 @@ const int *mbedtls_cipher_list( void );
  * \brief               This function retrieves the cipher-information
  *                      structure associated with the given cipher name.
  *
- * \param cipher_name   Name of the cipher to search for.
+ * \param cipher_name   Name of the cipher to search for. This can be \c NULL.
```

Comment:
I think we should require this to be non-NULL and make the existing check conditional.

Reply:
@hanno-arm I considered doing so, but I was afraid that there is software that depends on the `NULL` return. Also, this is neither `void` nor error code returning function, thus we don't have a common parameter validation scheme for this case. I could add a non-returning optional check, but then it still needs to return `NULL` upon receiving a `NULL` argument.

Do you have a specific idea, how to handle this case? I still think it may be a feasible option to just document the current behavior, but I am willing to handle it in another way.

Reply:
Good point on the return value! In that case, I'd suggest we keep the unconditional check, but still document that cipher_name must not be NULL.